### PR TITLE
Extend mappings interface and add v2 implementation of said interface

### PR DESCRIPTION
**Pull request by natanfudge**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ buildscript {
 }
 
 sourceCompatibility = 1.8
-version = '0.1.1'
+version = '0.1.1-local'
```
Comment (liach) [inline build.gradle]: Fyi maybe we will bump this later

File: src/main/java/net/fabricmc/tinyv2/V2MappingsProvider.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        private LocalVariable[] currentLocalVariableNames;
+        private List<String> currentComments;
+        private CommentType currentCommentType;
+
+        @Override
+        public void start(TinyMetadata metadata) {
+            namespaces = metadata.getNamespaces();
+            namespaceAmount = namespaces.size();
+            for (int i = 0; i < namespaceAmount; i++) {
+                namespaceIndices.put(namespaces.get(i), i);
```
Comment (liach) [inline src/main/java/net/fabricmc/tinyv2/V2MappingsProvider.java]: Consider replacing this map with a `ToIntFunction<String>`, which will output an `int` for a `String` input. So like
```java
private ToIntFunction<String> namespaceIndices;

// stuff

namespaceIndices = metadata::index;
```

File: src/test/java/net/fabricmc/tinyv2/MappingsInterfaceTest.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class MappingsInterfaceTest {
+    @Test
+    public void testParsingToOldInterface() throws IOException {
+        try (BufferedReader reader = Files.newBufferedReader(Paths.get("src/test/resources/bigboi_mappings.tinyv2"))) {
```
Comment (liach) [inline src/test/java/net/fabricmc/tinyv2/MappingsInterfaceTest.java]: This may break once the run directory changes. Should use classloader getResource

Comment (natanfudge) [reply src/test/java/net/fabricmc/tinyv2/MappingsInterfaceTest.java]: I tried that and it broke

Comment (natanfudge) [reply src/test/java/net/fabricmc/tinyv2/MappingsInterfaceTest.java]: Can have another try

**Pull request merged**